Add ability to use EDITOR env var as log view text editor

gregdenton opened the pull request:
Fixes Issue #179 
Priority for the log view editor is as follows:
* EDITOR env variable
* If not defined use the Constants.DEFAULT_EDITOR variable

This change also will handle errors in launching the editor by displaying an error message to the user.

bcipriano commented on cuegui/cuegui/Utils.py:
```diff
@@ -416,8 +442,10 @@ def popupTail(file, facility=None):
-        JOB_LOG_CMD = str(QtGui.qApp.settings.value("LogEditor", DEFAULT_EDITOR))
-        shellOut("%s %s" % (JOB_LOG_CMD or DEFAULT_EDITOR, file))
+        editor = os.getenv('EDITOR', DEFAULT_EDITOR).split()
+        job_log_cmd = QtGui.qApp.settings.value("LogEditor", editor) or editor
```
nit: I know it's not new but I'm not a huge fan of the `or` here - this looks like a three-choice series that would be more clear in an if-elif-else block.

gregdenton replied:
Yeah, this section felt a bit messy and looking at it again.
`job_log_cmd = QtGui.qApp.settings.value("LogEditor", editor) or editor`
The second parameter of the `value` method is the default value, so this `or` isn't doing anything anyways. I'm cleaning it up now.

gregdenton commented:
@bcipriano PTAL.
I cleaned this up a bit and also changed the resolution order. I feel the order should resolve from least persistent to most. So environment variable, qsettings file, default editor. Let me know if you agree.
